File: modules/core/src/model/gvl/Vendor.ts
import {GVLMapItem} from './GVLMapItem.js';

export interface Vendor extends GVLMapItem {
  purposes: number[];
  legIntPurposes: number[];
  flexiblePurposes: number[];
  specialPurposes: number[];
  features: number[];
  specialFeatures: number[];
  policyUrl: string;
  usesCookies: boolean;
  cookieMaxAgeSeconds: number | null;
  cookieRefresh: boolean;
  usesNonCookieAccess: boolean;
  deviceStorageDisclosureUrl?: string;
  deletedDate?: Date | string;
  overflow?: {
    httpGetLimit: 32 | 128;
  };
}

File: GDPR-Phase2-Java/src/main/java/com/gdpr/phase2/model/gvl/Vendor.java
package com.gdpr.phase2.model.gvl;


import lombok.*;

import java.util.*;

@Getter @Setter
class Overflow {
    private float httpGetLimit;
}
@Getter @Setter @AllArgsConstructor @NoArgsConstructor
public class Vendor extends GVLMapItem {
    List<Integer> purposes;
    List<Integer> legIntPurposes;
    List<Integer> flexiblePurposes;
    List<Integer> specialPurposes;
    List<Integer> features;
    List<Integer> SpecialFeatures;
    String policyUrl;
    Date deletedDate;
    Overflow overflow;



}
